Cython/Utility/ImportExport.c

```diff
@@ -71,6 +74,30 @@ static PyObject *__Pyx__ImportDottedModule_Lookup(PyObject *name) {
 }
 #endif
 
+#if PY_MAJOR_VERSION >= 3
+static PyObject *__Pyx_ImportDottedModule_WalkParts(PyObject *module, PyObject *name, PyObject *parts_tuple) {
+    Py_ssize_t i, nparts;
+    nparts = PyTuple_GET_SIZE(parts_tuple);
+    for (i=1; i < nparts && module; i++) {
+        PyObject *part, *submodule;
+#if CYTHON_ASSUME_SAFE_MACROS && !CYTHON_AVOID_BORROWED_REFS
+        part = PyTuple_GET_ITEM(parts_tuple, i);
+#else
+        part = PySequence_ITEM(parts_tuple, i);
+#endif
+        submodule = __Pyx_PyObject_GetAttrStrNoError(module, part);
```

Comment:
submodule could be `NULL` here I think, which isn't handled (except on the last loop iteration).

(I realise this is moved code rather than new code)

Reply:
It's handled in the loop above: `    for (i=1; i < nparts && module; i++) {`
Note that `module` is simply reassigned from `submodule` at the end of the loop without touching it any further.
You're right that this isn't obvious, though. I'll add a comment, thanks.